catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTableOperations.java

```diff
@@ -216,11 +220,60 @@ private static void appendPartitionSql(
           "The partition field must be one of the columns");
 
       String partitionColumn = BACK_QUOTE + rangePartition.fieldName()[0] + BACK_QUOTE;
-      // TODO we currently do not support pre-assign partition when creating range partitioning
-      partitionSqlBuilder.append(partitionColumn).append(") () ");
+      partitionSqlBuilder.append(partitionColumn).append(") ").append(NEW_LINE).append("(");
+
+      RangePartition[] assignments = rangePartition.assignments();
+      if (!ArrayUtils.isEmpty(assignments)) {
+        ImmutableList.Builder<String> partitions = ImmutableList.builder();
+        for (RangePartition part : assignments) {
+          StringBuilder partitionAssignSqlBuilder = new StringBuilder();
+          partitionAssignSqlBuilder
+              .append(" PARTITION ")
+              .append(BACK_QUOTE)
+              .append(part.name())
+              .append(BACK_QUOTE)
+              .append(" VALUES ");
+          LiteralDTO upper = (LiteralDTO) part.upper();
```

Comment:
I wonder why literal values of partitons here are instances of `LiteralDTO`, in my understanding, it seems like they should be instances of `LiteralImpl`, is this a mistake?

Reply:
yes, the mistake is due to the absence of a conversion in DTOConverters. 

Can you prioritize fixing this issue in a separate PR?

Reply:
please rebase the latest main branch and fix this